src/core/services/email.py

```diff
@@ -0,0 +1,83 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.core.db.db import get_session
+from src.core.db.repository.user import UserRepository
+from src.core.exceptions import exceptions
+from src.settings import settings
+
+
+class EmailSchema(BaseModel):
+    recipients: list[EmailStr]
+    template_body: dict[str, Any] | None
+
+
+class EmailProvider:
+    """Класс для отправки электронных писем."""
+
+    def __init__(self, sessionmaker: Generator[AsyncSession, None, None] = get_session):
+        conf = ConnectionConfig(
+            MAIL_USERNAME=settings.MAIL_LOGIN,
+            MAIL_PASSWORD=settings.MAIL_PASSWORD,
+            MAIL_FROM=settings.ORGANIZATIONS_EMAIL,
+            MAIL_PORT=settings.MAIL_PORT,
+            MAIL_SERVER=settings.MAIL_SERVER,
+            MAIL_FROM_NAME="ProCharity Bot",
+            MAIL_STARTTLS=settings.MAIL_STARTTLS,
+            MAIL_SSL_TLS=settings.MAIL_SSL_TLS,
+            USE_CREDENTIALS=settings.USE_CREDENTIALS,
+            VALIDATE_CERTS=settings.VALIDATE_CERTS,
+        )
+        self.fastmail = FastMail(conf)
+        self._sessionmaker = contextlib.asynccontextmanager(sessionmaker)
+
+    async def __send_mail(
+        self,
+        email_obj: EmailSchema,
+        subject: str,
+        template_name: str | None,
+        body: str | None,
+    ) -> None:
+        """Базовый метод отправки сообщения на электронную почту.
+        Аргументы:
+            recipients (list[EmailStr]): список email получателей
+            subject (str): тема сообщения
+            template_body (dict[str, Any]): значения переменных для шаблона сообщения
+            template_name (str): название шаблона для сообщения
+            body (str): тело электронного письма
+        """
+        message = MessageSchema(
+            subject=subject,
+            recipients=email_obj.recipients,
+            template_body=email_obj.template_body,
+            body=body,
+            subtype=MessageType.html,
+        )
+
+        try:
+            await self.fastmail.send_message(message, template_name)
+        except Exception as exc:
+            raise exceptions.EmailSendError(email_obj.recipients, exc)
+
+    async def send_question_feedback(self, telegram_id: int, message: str, email: list[EmailStr]) -> None:
+        """Отправляет email на почтовый ящик администратора с отзывом/вопросом."""
+        recipients = email
+        email_obj = EmailSchema(recipients=recipients, template_body=None)
+        try:
+            async with self._sessionmaker() as session:
+                user_repository = UserRepository(session)
+                user = await user_repository.get_by_telegram_id(telegram_id)
+        except ValueError:
+            user = None
+        if user.email is None:
+            user.email = "пользователь не указал свой email"
+        await self.__send_mail(
+            email_obj,
+            subject=f"Сообщение от пользователя {user.username}(email: {user.email})",
```

Comment:
Тут лучше использовать полное имя
```suggestion
            subject=f"Сообщение от пользователя {user.full_name} ({user.email or 'пользователь не указал свой email'})",
```

Reply:
Исправлю

Reply:
@NiKuma0 а зачем такая конструкция  - {user.email or 'пользователь не указал свой email'})"? у меня пишет сообщение, если почта не указана и при старой.  И еще одно уточнение - full_name это чтобы выводилось имя и фамилия? Я поля full_name не вижу у User.

Reply:
> Я поля full_name не вижу у User.

Если того поля нет, оставь просто `user.first_name`, потом добавим.

> а зачем такая конструкция - {user.email or 'пользователь не указал свой email'})"?

Это конструкция не то чтобы "нужна", так будет меньше вложенности, и, на мой взгляд ,проще будет читатся.

Reply:
Понял. Спасибо за разъяснения!